### PR TITLE
The great Spectral Blade permanent ghost trap unGBJing - 10/10 ghosts hated that sword

### LINKED ISSUE #48408: Spectral Swords force you to orbit forever
Issue reported from  Round ID: 126664 (/tg/Station Sybil [ENGLISH] [US-WEST] [100% LAG FREE])

Reporting client version: 512.1488

Have it in any round. Have your ghost orbit it, they will be stuck forever on the sword as attempting to orbit another person after that will just yank you back again.

### PR DESCRIPTION
**Pull request by Wayland-Smithy**
## About The Pull Request

Fixes a bug caused by transferring an orbiter component to a parent where one already exists. Causing the old original orbiter to update the added ghosts positions when the parent moves, but not allowing them to ever break orbit and also setting the ref to null for the component on the whole due to old signals it would seem.

Also tweaks the blades use in hand ghost notification message to remove the redundant `(Orbit)` link.

Fixes: #48408 

I would like a TM to make sure this change does not cause any other issues to crop up for the ghost gang.

#### Should bring an end to dchat messages like:
```
Bugger (DEAD): "is there a way i can stop orbiting ... the blade"
Terry Hunter (DEAD): "FCKING ANNOYING GHOST BLADE"
Wapons Shellar (DEAD): "why cant we stop orbiting the sword?"
Samuel Chilton (DEAD): "i fucking hate the ghost blade"
```

## Changelog
:cl:
fix: Spectral Blade - Ghosts that orbit it or the holder are no longer forced to spectate it until the end of time. Huzzah!
fix: For real. Not kidding. You can answer the call to aid someone with your spooky presence and then STOP ORBITING ON YOUR OWN AT ANY TIME just like any other object.
fix: Ghosts no longer receive a redundant orbit link when the Spectral Blade is raised.
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
oh thank god

**Comment by LemonInTheDark** [comment]
You might wanna make the cl even more clear, people are distrustful of this thing, need to make it as clear as possible that it's gone

**Comment by Ryll-Ryll** [comment]
It's honestly amazing that it took something like three years for this to be fixed.

**Comment by TemporalOroboros** [comment]
Does this work on #52615 by any chance?

**Comment by Wayland-Smithy** [comment]
Unfortunately this does not fix the issue with the blade stealing the wisp. Because the blade looks for all orbiters of all atoms in the same turf as it, and steals them using very generic component transfer functions I do not see a simple way to prevent the wisp in its current form from being taken just like any other ghost in the orbiters list.

I did find a workaround is to just keep a hold of the lantern, and you can bottle and re-release the wisp to have it orbit you again once the sword is gone.

> I guess it only got reported at the end of 2019

From what I saw it looks like it could have been happening since at least September 2018. I am just so glad to get off Mr. Blade's wild ride.

**Comment by Wayland-Smithy** [comment]
Converting to draft until the fix for a persisting issue with the orbiters atom var of the sword becoming null, still totally breaking its gimmick, is in.

File: code/datums/components/orbiter.dm [hunk 1]
```diff
@@ -25,24 +25,34 @@
...
 	return ..()
 
 /datum/component/orbiter/InheritComponent(datum/component/orbiter/newcomp, original, atom/movable/orbiter, radius, clockwise, rotation_speed, rotation_segments, pre_rotation)
 	if(!newcomp)
+		RegisterSignal(parent, COMSIG_MOVABLE_UPDATE_GLIDE_SIZE, .proc/orbiter_glide_size_update)
 		begin_orbit(arglist(args.Copy(3)))
 		return
 	// The following only happens on component transfers
+	for(var/atom/movable/incoming_orbiter in newcomp.orbiters)
+		incoming_orbiter.orbiting = src
```
Comment (Rohesie) [inline code/datums/components/orbiter.dm:49]:
```suggestion
	for(var/o in newcomp.orbiters)
		var/atom/movable/incoming_orbiter = o
		incoming_orbiter.orbiting = src
```
`orbiters` could really stand not to be the same name for two very different things.

**Comment by Rohesie** [comment]
Is this ready and tested locally?

**Comment by Wayland-Smithy** [comment]
Ready and tested locally with these repro steps:

1. Join dev server, and spawn in as any job
2. Game Panel spawn a `wisp_lantern` and a `ghost_sword`, but do not pickup or stand on the sword
3. Use the lantern to gain a orbiter on yourself then transfer it to the blade by walking over it
4. Ghost and orbit yourself, then use build mode to throw your cata self onto the sword, now 2 orbiters have been transferred to the sword and have properly "merged"
5. Break orbit when your ghost, and when the sword moves you will no longer be stuck following it forever and the orbiters atom var is no longer prematurely set to null

**Comment by Wayland-Smithy** [comment]
Ready and tested locally with these repro steps, which are a little easier than those I posted on the issue:

1. Join dev server, and spawn in as any job
2. Game Panel spawn a `wisp_lantern` and a `ghost_sword`, but do not pickup or stand on the sword
3. Use the lantern to gain a orbiter on yourself then transfer it to the blade by walking over it
4. Ghost and orbit yourself, then use build mode to throw your cata self onto the sword, now 2 orbiters have been transferred to the sword and have properly "merged"
5. Break orbit with your ghost, and when the sword moves you will no longer be stuck following it forever and the orbiters atom var is no longer prematurely set to null

**Comment by Rohesie** [comment]
Great work, kudos.

**Pull request merged**